### PR TITLE
refactor(consensus): The consensus status now always represents the current status

### PR DESCRIPTION
**Pull request by yejiayu**
Testing in DevOps.

**What type of PR is this?**
refactor

**What this PR does / why we need it**:
**Rename**
- rename `status.xx_root` to `status.list_xx_root`
- rename `status.update_after_exec ` to `status.update_by_executed`
- rename `status.update_after_commit ` to `status.update_by_commited`
- `height` and `prevhash` have been modified to fit the current semantics.

**Refactor**
- Synchronization now shares the latest height with status and is no longer available from storage.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: core/consensus/src/status.rs [hunk 1]
```diff
@@ -85,23 +68,23 @@ impl StatusAgent {
...
-    pub prev_hash:          Hash,
+    pub current_hash:       Hash,
     pub latest_state_root:  MerkleRoot,
-    pub logs_bloom:         Vec<Bloom>,
-    pub confirm_root:       Vec<MerkleRoot>,
-    pub state_root:         Vec<MerkleRoot>,
-    pub receipt_root:       Vec<MerkleRoot>,
-    pub cycles_used:        Vec<u64>,
-    pub proof:              Proof,
+    pub list_logs_bloom:         Vec<Bloom>,
```
Comment (KaoImin) [inline core/consensus/src/status.rs]: This struct skips `rustfmt`, it should be formatted manually.

File: core/consensus/src/status.rs [hunk 2]
```diff
@@ -295,7 +204,7 @@ impl CurrentConsensusStatus {
     fmt = "exec height {}, cycles used {}, state root {:?}, receipt root {:?}, confirm root {:?}, logs bloom {}",
     exec_height, cycles_used, state_root, receipt_root, confirm_root, "logs_bloom.to_low_u64_be()"
 )]
-pub struct UpdateInfo {
+pub struct ExecutedInfo {
     pub exec_height: u64,
```
Comment (KaoImin) [inline core/consensus/src/status.rs]: ditto

**Comment by KaoImin** [comment]
LGTM

File: core/consensus/src/engine.rs [hunk 3]
```diff
@@ -47,41 +47,44 @@ impl<Adapter: ConsensusAdapter + 'static> Engine<FixedPill> for ConsensusEngine<
     async fn get_block(
         &self,
         ctx: Context,
-        height: u64,
+        next_height: u64,
```
Comment (rev-chaos) [inline core/consensus/src/engine.rs]: next_height may not be appropriate. For consensus, it gets current_height's block

File: core/consensus/src/engine.rs [hunk 4]
```diff
@@ -153,7 +156,7 @@ impl<Adapter: ConsensusAdapter + 'static> Engine<FixedPill> for ConsensusEngine<
     async fn commit(
         &self,
         ctx: Context,
-        height: u64,
+        current_height: u64,
```
Comment (rev-chaos) [inline core/consensus/src/engine.rs]: name by consensus_height may more fittable

**Comment by yejiayu** [comment]
Replaced `latest_state_root` with` latest_commited_state_root`

`latest_commited_state_root` will always record the state of the last commit, ensuring that the state will not be lost if the node performs inconsistently.

ping @KaoImin @rev-chaos

**Pull request merged**